deps/chakrashim/src/jsrtutils.cc

```diff
@@ -1061,5 +1061,47 @@ JsValueRef CALLBACK CollectGarbage(
   return jsrt::GetUndefined();
 }
 
+void IdleGC(uv_prepare_t *prepareHandle) {
+  unsigned int nextIdleTicks;
+  CHAKRA_VERIFY(JsIdle(&nextIdleTicks) == JsNoError);
+  DWORD currentTicks = GetTickCount();
+
+  // remove the idleGC from prepareQueue for now.
+  uv_prepare_stop(prepareHandle);
+
+  // If idleGc completed, we don't need to schedule anything.
+  // If idleGc didn't complete, runtime will try to do it after nextIdleTicks.
+  // In that case, add idleGc callback in prepareQueue after
+  // (nextIdleTicks - currentTicks) ms.
+  if (nextIdleTicks != UINT_MAX) {
+    unsigned int diff = nextIdleTicks - currentTicks;
+    if (diff > 0) {
+      StartTimerIdleGCScheduler(diff);
+    }
+  }
+}
+
+
+void ScheduleIdleGC(uv_timer_t* timerHandler) {
+  // add idleGC callback into prepareQueue
+  uv_prepare_start(IsolateShim::GetCurrent()->idleGc_prepare_handle(), IdleGC);
+
+  // stop the scheduler (for now).
+  uv_timer_stop(IsolateShim::GetCurrent()->idleGc_timer_handle());
```

Comment:
does the order matter? should we stop the old timer before trying to schedule a new one?